common/test/acceptance/pages/studio/course_info.py

```diff
@@ -1,16 +1,150 @@
+
+    def submit_update(self, message):
+        """
+        Adds update text to the new update CodeMirror form and submits that text
+
+        Arguments:
+            message (str): The message to be added and saved
+        """
+        type_in_codemirror(self, 0, message)
+        self.click_new_update_save_button()
+
+    def set_date(self, date):
+        """
+        Sets the updates date input to the provided value
+
+        Arguments:
+            date (str): Date string in the format DD/MM/YYYY
+        """
+        set_input_value(self, 'input.date', date)
+
+    def is_update_date(self, search_date):
+        """
+        Checks to see if the search date is present
+
+        Arguments:
+            search_date (str): e.g. 06/01/2013 would be found with June 1, 2013
+        """
+        dates = self.q(css='.date-display').html
+        for date in dates:
+            if search_date == date:
+                return True
+        return False
+
+    def is_new_update_save_button_present(self):
+        """
+        Checks to see if the CodeMirror Update save button is present
+        """
+        return self.q(css='.save-button').present
+
+    def click_new_update_save_button(self):
+        """
+        Clicks the CodeMirror Update save button
+        """
+        click_css(self, '.save-button')
+
+    def is_edit_button_present(self):
+        """
+        Checks to see if the edit update post buttons if present
+        """
+        return self.q(css='div.post-preview .edit-button').present
+
+    def click_edit_update_button(self):
+        """
+        Clicks the edit update post button
+        """
+        # self.q(css='div.post-preview .edit-button').first.click()
+        click_css(self, 'div.post-preview .edit-button', require_notification=False)
+        self.wait_for_element_visibility('.CodeMirror', 'Waiting for .CodeMirror')
+
+    def is_delete_update_button_present(self):
+        """
+        Checks to see if the delete update post button is present
+        """
+        return self.q(css='div.post-preview .delete-button').present
+
+    def click_delete_update_button(self):
+        """
+        Clicks the delete update post button
+        """
+        click_css(self, 'div.post-preview .delete-button', require_notification=False)
+        confirm_prompt(self)
+
+    def is_saving_deleting_notification_present(self):
+        """
+        Checks for the presence of the saving or deleting notification
+        """
+        return self.q(css='#notification-mini').present
+
+    def update_text_contains(self, message):
+        """
+        Looks for the message in the list of updates in the page
+
+        Arguments:
+            message (str): String containing the message that is to be searched for
+
+        Returns:
+            bool: True if an instance of the message is found
+        """
+        update = self.q(css='div.update-contents').html
+        if len(update) > 0:
```

Comment:
This is problematic though because the method could return False either because there are no updates on the page, or because the text you pass in does not match the message in the first update on the page. You should separate these two concerns so that there is a method that specifically checks that there are no updates on the page.